runtime/include/chpl-topo.h

```diff
@@ -58,6 +58,23 @@ int chpl_topo_getNumCPUsLogical(chpl_bool /*accessible_only*/);
 //
 int chpl_topo_getCPUs(chpl_bool physical, int *cpus, int count);
 
+// Reserves a physical CPU (core) and returns its hwloc OS index. The
+// core and its PUs will no longer be included in the results returned
+// by chpl_topo_getCPUs, chpl_topo_getNumCPUsPhysical, and
+// chpl_topo_getNumCPUsLogical.
+//
+// Returns -1 if the reservation failed.
+//
+int chpl_topo_reserveCPUPhysical(void);
```

Comment:
We probably need this to be called before `chpl_topo_getCPUs*` is called. If that's true, I wonder we can add a check to verify we get the call order right.

Reply:
Good suggestion.

Reply:
There are a couple of issues with this. I'm assuming that core reservation also must be done before calling `chpl_topo_getNumCPUs*`, which relies on the same information. `chpl_comm_regMemHeapTouch` calls `chpl_topo_getNumCPUsPhysical` to get the number of cores to use when faulting in the heap. Is it ok if it doesn't use the reserved core(s) to do the faulting?

Reply:
Yeah, I think so. That code is just interleaving pages, and I'm pretty sure isn't sensitive to the total number of cores. And losing a core shouldn't really impact the speed since almost any modern CPU can outdrive the memory bandwidth